darts/models/regression_model.py

```diff
@@ -13,300 +13,259 @@
+            self.lags_exog = self.lags[:]
+        elif self.lags_exog is False:
+            self.lags_exog = [0]
+        elif isinstance(self.lags_exog, int):
+            raise_if_not(self.lags_exog > 0, "`lags_exog` must be strictly positive. Given: {}.".format(self.lags_exog))
+            self.lags_exog = list(range(1, self.lags_exog+1))
+        elif isinstance(self.lags_exog, list):
+            for lag in self.lags_exog:
+                raise_if(not isinstance(lag, int) or (lag < 0), (
+                    "Every element of `lags_exog` must be a positive integer. Given: {}."
+                    .format(self.lags_exog))
+                )
+
+        self.model = model
+        if self.lags is not None and self.lags_exog is None:
+            self.max_lag = int(np.max(self.lags))
+        elif self.lags is None and self.lags_exog is not None:
+            self.max_lag = int(np.max(self.lags_exog))
+        else:
+            self.max_lag = int(np.max([np.max(self.lags), np.max(self.lags_exog)]))
         self._fit_called = False
 
-    @abstractmethod
-    def fit(self, train_features: List[TimeSeries], train_target: TimeSeries) -> None:
+    def fit(self, series: TimeSeries, exog: Optional[TimeSeries] = None, **kwargs) -> None:
         """ Fits/trains the model using the provided list of features time series and the target time series.
 
         Parameters
         ----------
-        train_features
-            A list of features time series, all of the same length as the target series
-        train_target
-            A target time series, of the same length as the features series
+        series : TimeSeries
+            TimeSeries object containing the target values.
+        exog : TimeSeries, optional
+            TimeSeries object containing the exogenous values.
         """
+        super().fit(series, exog)
+        raise_if(exog is not None and self.lags_exog is None,
+            "`exog` not None in `fit()` method call, but `lags_exog` is None in constructor. "
+        )
+        raise_if(exog is None and self.lags_exog is not None,
+            "`exog` is None in `fit()` method call, but `lags_exog` is not None in constructor. "
+        )
+        self.target_column = series.columns()[0]
+        self.exog_columns = exog.columns().values.tolist() if exog is not None else None
+        if self.exog_columns is not None:
+            if str(self.target_column) in self.exog_columns:
+                series = TimeSeries(series.pd_dataframe().rename({self.target_column: "Target"}))
+                self.target_column = series.columns()[0]
+        self.training_data = self._create_training_data(series, exog)
+        self.train_x =  self.training_data.pd_dataframe().drop(self.target_column, axis=1)
+        self.train_y =  self.training_data.pd_dataframe()[self.target_column]
+        self.nr_exog = exog.width*len(self.lags_exog) if exog is not None else 0
+
+        self.model.fit(
+            X=self.train_x,
```

Comment:
This assumes that the arguments are name `X` and `y`. Maybe that's fine (for sure it'll work for sklearn), but perhaps we could even make it more "agnostic" and not used named parameters here?

Reply:
You're right. Removed the named attribute.